employees/views.py

```diff
@@ -154,6 +156,80 @@ def redirect_to_current_month(self) -> HttpResponseRedirectBase:
         return redirect(self._get_recent_month_url(pk))
 
 
+@method_decorator(login_required, name="dispatch")
+@method_decorator(
+    check_permissions(
+        allowed_user_types=[
+            CustomUser.UserType.EMPLOYEE.name,
+            CustomUser.UserType.MANAGER.name,
+            CustomUser.UserType.ADMIN.name,
+        ]
+    ),
+    name="dispatch",
+)
+class ReportListCreateProjectJoinView(MonthNavigationMixin, CreateView):
+    template_name = "employees/report_list.html"
+    project_join_form = ProjectJoinForm
+    model = Report
+    form_class = ReportForm
+    url_name = "custom-report-list"
+    object = None
+
+    def get_initial(self) -> dict:
+        initial = super().get_initial()
+        initial.update({"date": timezone.now(), "user": self.request.user})
+        return initial
+
+    def form_valid(self, form: ReportForm) -> HttpResponseRedirectBase:
+        self.object = form.save(commit=False)
+        self.object.author = self.request.user
+        self.object.full_clean()
+        self.object.save()
+        return redirect(self.get_success_url())
+
+    def get_queryset(self) -> QuerySet:
+        return (
+            super()
+            .get_queryset()
+            .filter(author=self.request.user, date__year=self.kwargs["year"], date__month=self.kwargs["month"])
+            .order_by("-date", "project__name", "-creation_date")
+        )
+
+    def get_context_data(self, **kwargs: Any) -> dict:
+        context_data = super().get_context_data(**kwargs)
+        context_data["UI_text"] = ReportListStrings
+        context_data["object_list"] = self.get_queryset()
+        context_data["daily_hours_sum"] = context_data["object_list"].order_by().get_work_hours_sum_for_all_dates()
+        context_data["monthly_hours_sum"] = context_data["object_list"].order_by().get_work_hours_sum_for_all_authors()
+        context_data["project_form"] = ProjectJoinForm(
+            queryset=Project.objects.exclude(members__id=self.request.user.id).order_by("name")
+        )
+        context_data["hide_join"] = not Project.objects.exclude(members__id=self.request.user.id).exists()
```

Comment:
Yes, it would be because currently there are two database queries, and it can be done with one. However, this would require modifying `ProjectJoinForm` to accept list instead of queryset. Then if you evaluate first one (by making it into a `list()`) and assign it to variable, you can pass variable to form. Then in line 207 you can check if list is not empty. 

But as I have said, it would require modifying how `ProjectJoinForm` is instantiated here and in other places, so if we have some-kind-of standard that we pass queryset to forms, we would break it.

Still nice catch!